backend/data/src/main/kotlin/io/tolgee/model/UserAccount.kt

```diff
@@ -80,6 +79,18 @@ data class UserAccount(
 
   override var avatarHash: String? = null
 
+  @Column(name = "deleted_at")
+  var deletedAt: Date? = null
+
+  val isDeletable: Boolean
+    get() = this.accountType != AccountType.LDAP
+
+  val isMfaEnabled: Boolean
+    get() = this.totpKey?.isNotEmpty() ?: false
+
+  val needsSuperJwt: Boolean
+    get() = this.accountType != AccountType.THIRD_PARTY || isMfaEnabled
```

Comment:
Maybe we should enable users to set MFA without setting password? Also Maybe it would be good idea to remove the accoutType field from the DB and provide just getter which will return the `if(this.password != null) AccountType.Local else if(this.thirdPartyId) AccountType.THIRD_PARTY else AccountType.LDAP`?

Reply:
> Maybe we should enable users to set MFA without setting password?

I feel like there's no harm in requiring them to have a password to enable MFA, as they still can login via OAuth. There's also the bug you pointed out on Slack that needs to be looked at regaring OAuth & MFA.

> Maybe it would be good idea to remove the accoutType field from the DB and provide just getter

I generally prefer duplicate data and data that could be computed to be directly available in the DB rather than computed values when it makes sense; Disk space is much cheaper than CPU power and this is a recommended thing in many NoSQL databases which is the world I'm coming from with Mongo where we prefer to duplicate data rather than having to aggregate (aka `JOIN` in SQL terms) 😄

Both are equally valid, and I doubt a simple check will draw crazy amounts of CPU, so I guess it's up to you to change that to a computed field if you prefer.

----

Going back to the MFA point; If we want to securely enable MFA we'd have to come up with some form of email code, which means we'd have a way to reauthenticate users which do not have a password or MFA; The `needsSuperJwt` would therefore become useless as it'd always be `true`